Lib/enum.py

```diff
@@ -130,7 +144,21 @@ def __new__(metacls, cls, bases, classdict):
         # cannot be mixed with other types (int, float, etc.) if it has an
         # inherited __new__ unless a new __new__ is defined (or the resulting
         # class will fail).
-        #
+
+        # Get __qualname__ for the class being created.
+        enum_class_qualname = super().__new__(metacls, cls, bases, classdict).__qualname__
+
+        # We want to avoid treating locally-defined nested classes as enum
+        # values, so we use __qualname__ to determine this.
+        # e.g. if a class Bar is defined inside an enum Foo, then say if
+        # enum_class.__qualname__ is Foo, then member.__qualname__ will be Bar.
+        # We have to do it here since __qualname__ of the new Enum isn't
+        # accessible in _EnumDict.__setitem__().
+        for key, member in classdict.items():
+            if isinstance(member, type):
+                if member.__qualname__.startswith(enum_class_qualname):
```

Comment:
This catches
```
class MyEnumInner(object):
    pass

class MyEnum(Enum):
    oops = MyEnumInner
```

Should be `.startswith(enum_class_qualname + '.')`

Reply:
Thanks again - just fixed.